pyccel/ast/internals.py

```diff
@@ -245,6 +245,103 @@ def code(self):
         """
         return self._code
 
+#------------------------------------------------------------------------------
+class CStringExpression(Basic):
+    """
+    Internal class used to hold a C string that has LiteralStrings and C macros
+    """
+    __slots__  = ('_expression',)
+    _attribute_nodes  = ()
+
+    def __init__(self, *args):
+        super().__init__()
+        self._expression = []
+        for arg in args:
+            self += arg
+
+    def __repr__(self):
+        return ''.join([str(e) for e in self._expression])
```

Comment:
Should be `return ''.join(repr(e) for e in self._expression)` . See [here](https://www.geeksforgeeks.org/str-vs-repr-in-python/) for the difference between `str` and `repr`